### PR TITLE
[Fundamental] Introduce stale for issue/PR status monitor

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

This PR targets to leverage [actions/stale](https://github.com/actions/stale) to add label for stale issue/PR.

In short, issue will be tagged label `no-recent-activity` after 30 days, and stale PR will be tagged after 7 days. Issues with label `long-term` will be excluded.

**References**

- Azure SDK stale issue: <https://github.com/Azure/azure-sdk-for-python/issues/32959>
- Azure SDK stale pull request: <https://github.com/Azure/azure-sdk-for-python/pull/32182>

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      issues: write
+      pull-requests: write
+    steps:
+      - uses: actions/stale@v8
+        with:
+          stale-issue-message: 'This issue is stale because it has been open 30 days with no activity. Remove stale label or comment or this will be closed in 5 days.'
+          stale-pr-message: 'This PR is stale because it has been open 45 days with no activity. Remove stale label or comment or this will be closed in 10 days.'
+          close-issue-message: 'This issue was closed because it has been stalled for 5 days with no activity.'
+          close-pr-message: 'This PR was closed because it has been stalled for 10 days with no activity.'
+          days-before-issue-stale: 30
```
Comment (zhengfeiwang) [inline .github/workflows/stale.yml:19]: @wangchao1230 let's make a decision on this configuration value

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: 'Close stale issues and PRs'
+on:
+  schedule:
+    - cron: '30 1 * * *'
```
Comment (crazygao) [inline .github/workflows/stale.yml:4]: add explain on cron time

File: .github/workflows/stale.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      - uses: actions/stale@v8
+        with:
+          stale-issue-message: 'This issue is stale because it has been open 30 days with no activity. Remove stale label or comment or this will be closed in 5 days.'
+          stale-pr-message: 'This PR is stale because it has been open 45 days with no activity. Remove stale label or comment or this will be closed in 10 days.'
+          close-issue-message: 'This issue was closed because it has been stalled for 5 days with no activity.'
+          close-pr-message: 'This PR was closed because it has been stalled for 10 days with no activity.'
+          days-before-issue-stale: 30
+          days-before-pr-stale: 45
+          days-before-issue-close: 5
+          days-before-pr-close: 10
```
Comment (wangchao1230) [inline .github/workflows/stale.yml:22]: image
we might refine the wording following this. Will the auto message at the contributor?

File: .github/workflows/stale.yml (hunk 1 above)
Comment (wangchao1230) [reply .github/workflows/stale.yml:19]: Can we follow this? 
image

File: .github/workflows/stale.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      pull-requests: write
+    steps:
+      - uses: actions/stale@v8
+        with:
+          stale-issue-message: 'This issue is stale because it has been open 30 days with no activity. Remove stale label or comment or this will be closed in 5 days.'
```
Comment (wangchao1230) [inline .github/workflows/stale.yml:15]: 7 days here.

File: .github/workflows/stale.yml (hunk 3 above)
Comment (zhengfeiwang) [reply .github/workflows/stale.yml:22]: I think it's doable, I notice there are some macros seems good to this.

Comment (zhengfeiwang) [reply .github/workflows/stale.yml:22]: maybe I can do some test in my repo first, I think we cannot afford touching all issues in this repo with unpredict behavior

File: .github/workflows/stale.yml (hunk 4 above)
Comment (zhengfeiwang) [reply .github/workflows/stale.yml:15]: so we will add stale label after 7 days, for both issue and PR?

File: .github/workflows/stale.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+          stale-pr-message: 'This PR is stale because it has been open 45 days with no activity. Remove stale label or comment or this will be closed in 10 days.'
```
Comment (wangchao1230) [inline .github/workflows/stale.yml:16]: can we add special treatment for issues marked with long-term?

Comment (zhengfeiwang) [reply .github/workflows/stale.yml]: yes, it is configurable with arg `exempt-issue-labels`, here is my test:

- long-term issue: <https://github.com/zhengfeiwang/gh-actions/issues/7>
- not long-term issue: <https://github.com/zhengfeiwang/gh-actions/issues/8>

File: .github/workflows/stale.yml (hunk 4 above)
Comment (zhengfeiwang) [reply .github/workflows/stale.yml]: got it, updated

File: .github/workflows/stale.yml (hunk 3 above)
Comment (zhengfeiwang) [reply .github/workflows/stale.yml]: have tested, unfortunately, the mention is a feature implemented inside Azure SDK tools: may take a glance at [this](https://github.com/Azure/azure-sdk-tools/blob/1e06e970fa28551411ea2ce1f596273d4ef14275/tools/github-event-processor/RULES.md?plain=1#L329)

File: .github/workflows/stale.yml (hunk 2 above)
Comment (zhengfeiwang) [reply .github/workflows/stale.yml]: added.

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request merged**